### ISSUE TITLE
Verifiable by default when Strict mocking

### ISSUE THREAD

**Issue by jezzsantos** [florinn/typemoq#57]
If using the `MockBehavior.Strict` mocking pattern (i.e. define all setups and then call `verifyAll()` at the end) the intention is that no code of the mocked type should be called unless there is explicit and matching a setup defined for it. `Strict` to most developers implies 'verbose', since it more tightly couples the tests the code.

At present, TypeMoq in `Strict` mode still forces the developer to call `mock.verifiable()` as well.

So given this class:

```
class Cut {

    do(dependency:Dependency):number {
        return dependency.something();
    }

    fetch(value:AClass) {
        return value.property;
    }
}

class Dependency {
    something():number {
        return 1;
    }
}
```
You still have to call `.verifiable()` for TypeMoq to ensure the `setup` expectation is called:

```
describe('simple relationship', function () {
            let dependency:TypeMoq.IMock<Dependency> = TypeMoq.Mock.ofType(Dependency, TypeMoq.MockBehavior.Strict);
            let cut:Cut = new Cut();

            it('cut delegates to dependency', () => {
                dependency.setup(dep => dep.something())
                    .returns(() => 1)
                    .verifiable(TypeMoq.Times.once());  //can get a false positive if this line is not included

                const result = cut.do(dependency.object);

                expect(result).to.equal(1);
                dependency.verifyAll();
            });
});
```

If you forget to call `.verifiable()` on the setup, AND the cut does not call the `dependency.something()` method that was expected, you get a false positive! 

Usability issues:
===

1. It is easy to forget to call `.verifiable()` especially when you have lots of setups. As required when using `Strict` mode, especially with code that has several dependencies.
2. `.verifiable()` is by default: `.verifiable(TypeMoq.Times.atLeastOnce())`. But in `Strict` mode, we are already opting-in to be strict (and verbose), and 99% of the time the most useful and most precise default is `.verifiable(TypeMoq.Times.once())`. Which is also ton of extra typing for the majority use-case, especially with a bunch of setups.
3. `Strict` mode, should be _Strict_ and unforgiving and precise. But currently, if you forget the call to `.verifiable()` at all, Strict is rendered very _Weak_ indeed, and that can lead to false positives that are hard to identify and debug.

Suggestion
===

What if, when using `Strict` mode on a mock instance, that every call to `.setup()` automatically called `.verifiable(Times.once())` for you, behind the scenes, as the default?

This frees the developer from having to make the explicit call: `.verifiable(TypeMoq.Times.once())` at all!

Then the developer can have the option to override the default `.verifiable()` by explicitly calling `.verifiable(TypeMoq.Times.whatever())` to get away from the default behavior.

This would not break existing code.
It does mean update to the docs, since using `Strict` would now mean `.verifiable()` is called for you and the default is `once()` not 'atLeastOnce()' by default.

It would mean that in `Strict` mode, developers can forget to call `.verifiable()` and not be penalized with false positives in TDD, AND developers don't have to type: `.verifiable(TypeMoq.Times.once())` so many times in a test file, just adding noise.

If developers are not using `Strict` then there is no change for them, `Loose` means loose, and the defaults are appropriate in the 'loose' testing pattern (i.e. create only required `setup`s with `It.isAny()` values, then write specific `.verify(whatever, TypeMoq.Times.whatever())` statements for only the stuff you care about).

**Comment by florinn** [comment]
I think you made a good case for changing the way verifiable works, thanks for the suggestion.

Please install the dev version and let me know if something is missing.

**Comment by jezzsantos** [comment]
Also works just as advertised. thanks again @florinn.

**Comment by florinn** [comment]
Thanks @jezzsantos . I've published a new version (v1.6.0) that includes these changes and those for #60 .
